addons/point_of_sale/views/pos_payment_views.xml

```diff
@@ -11,8 +11,14 @@
                         <field name="pos_order_id"/>
                         <field name="amount" readonly="0"/>
                         <field name="payment_method_id"/>
-                        <field name="card_type" invisible="not card_type"/>
-                        <field name="cardholder_name" invisible="not cardholder_name"/>
+                        <field name="payment_method_payment_mode" invisible="not payment_method_payment_mode"/>
+                        <field name="card_type" invisible="not card_type or card_no == 'null'"/>
```

Comment:
I'm not sure the `== 'null'` will work since you compare something to the String 'null' not the null value but if you are sure this works and tested it that is ok for me

Reply:
Hello @adgu-odoo 
we need to check for 'null' values for payments made by other methods like UPI. If we did not check for null value in card details then it will show null value in form view for payment like UPI